### PR TITLE
Run as non-root

### LINKED ISSUE #196: Run as non-root
Currently our Docker containers are running using root in production and dev. This used to be required for the Action Docker images and they share the same base image, so it stopped us from updating other services. However, these have since been updated and no longer require root privileges, which means it should be quite straightforward to update our remaining Docker images, most of which run internet-facing services.

This ticket is to change the user that's used to run the Actions Registry to a non-root user.

[Link to the risk in the Risk Register](https://docs.google.com/spreadsheets/d/14p4zoMhV5iAPTjzfTHaHDYBzkZ9Giw2yLUMTXCJdJFU/edit#gid=0&range=A30)

[Link to the change in Job Server to run as non-root](https://github.com/opensafely-core/job-server/commit/d76d26796a69b44a17b640be0870ce20be557fe8)

### PR DESCRIPTION
**Pull request by iaindillingham**
Closes #196

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -154,6 +154,9 @@ LABEL org.opencontainers.image.created=$BUILD_DATE
 ARG GITREF=unknown
 LABEL org.opencontainers.image.revision=$GITREF
 
+ARG USERID=10001
+ARG GROUPID=10001
```
Comment (madwort) [inline docker/Dockerfile:158]: hmm, so `10001` is the user id for job-server, defined [here](https://github.com/ebmdatalab/sysadmin/blob/011eda3ae139da08cd7259a06ba78f9790cfae90/infra/job-server/tasks.yaml#L12-L23)

reports uses `10002` defined [here](https://github.com/ebmdatalab/sysadmin/blob/011eda3ae139da08cd7259a06ba78f9790cfae90/infra/reports/tasks.yaml)

not sure which uid we should use for this.

Comment (madwort) [reply docker/Dockerfile:158]: actually, looking at `dokku3:/etc/passwd` we don't have a `10001` on that server, so we should change this. 

```
reports:x:10002:10002:Dedicated user for reports service to run as:/home/reports:/bin/sh
opencodelists:x:10003:10003:Dedicated user for opencodelists service to run as:/home/opencodelists:/bin/sh
ebmbot:x:10004:10004:Dedicated user ebmbot for bennettbot service to run as:/home/ebmbot:/bin/sh
```

Comment (madwort) [reply docker/Dockerfile:158]: see also https://github.com/ebmdatalab/sysadmin/pull/308/files

Comment (rebkwok) [reply docker/Dockerfile:158]: See https://github.com/ebmdatalab/sysadmin/pull/311
When this is merged, you'll just need to add an extra set of items for this user

**Comment by rebkwok** [comment]
Does actions-registry write anything (other than the db, which I assume is in a mounted volume in the dokku app)?  This was the thing that caused most problems for opencodelists. If everything it needs write access to is in a mounted storage volume, the sysadmin ansible playbook will do the work of making sure the non-root user owns it.

File: docker/Dockerfile (hunk 1 above)
Comment (iaindillingham) [reply docker/Dockerfile]: I've added a user/group for actions-registry on dokku3 and updated `docker/Dockerfile`. See ebmdatalab/sysadmin#322.

**Comment by iaindillingham** [comment]
Thanks for your comments, @madwort and @rebkwok. I've added a user/group for actions-registry on dokku3 and updated `docker/Dockerfile`. See ebmdatalab/sysadmin#322.

> Does actions-registry write anything?

Other than the DB, no, it doesn't.

File: docker/Dockerfile [hunk 2]
```diff
@@ -154,6 +154,9 @@ LABEL org.opencontainers.image.created=$BUILD_DATE
 ARG GITREF=unknown
 LABEL org.opencontainers.image.revision=$GITREF
 
+ARG USERID=10004
+ARG GROUPID=10004
```
Comment (rebkwok) [inline docker/Dockerfile:158]: Sorry, this uid is taken (by ebmbot); 10005 should be OK

**Review by rebkwok — CHANGES_REQUESTED** [review]

File: docker/Dockerfile (hunk 2 above)
Comment (iaindillingham) [reply docker/Dockerfile]: Thanks, @rebkwok. I've updated to 10005.

**Review by rebkwok — APPROVED** [review]

**Pull request merged**